src/web/lib/components/ThemeColorsEditor/index.js

```diff
@@ -9,21 +9,58 @@ import Metrics from "../../../../lib/metrics";
 import "./index.scss";
 
 class ThemeColorsEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selected: false,
+      colorName: null
+    };
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+  }
+
+  handleClick(name) {
+    const { setSelectedColor } = this.props;
+    setSelectedColor({ name });
+    if (this.state.selected && this.state.colorName !== name) {
+      this.setState({selected: this.state.selected, colorName: name });
+    } else {
+      this.setState({ selected: !this.state.selected, colorName: name });
+    }
+  }
+
   handleClickOutside() {
     const { selectedColor, setSelectedColor } = this.props;
     if (selectedColor !== null) {
       setSelectedColor({ name: null });
     }
+    this.setState({ selected: false, colorName: null });
+  }
+
+  handleKeyPress(event) {
+    if (event.keyCode === 27) {
```

Comment:
Same comment as above. Or since we're using it in two places now, not sure if there is a better place for shared constants.

Reply:
I guess I should put it in lib/constants.js then